plugins/polio/api.py

```diff
@@ -2082,6 +2091,230 @@ def get_queryset(self):
             )
 
 
+class CountryForVaccineSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = OrgUnit
+        fields = ["id", "name"]
+
+
+class CountryField(serializers.PrimaryKeyRelatedField):
+    def to_representation(self, value):
+        country = OrgUnit.objects.get(pk=value.pk)
+        serializer = CountryForVaccineSerializer(country)
+        return serializer.data
+
+
+class VaccineAuthorizationSerializer(serializers.ModelSerializer):
+    country = CountryField(queryset=OrgUnit.objects.filter(org_unit_type__name="COUNTRY"))
+
+    class Meta:
+        model = VaccineAuthorization
+        fields = [
+            "id",
+            "country",
+            "expiration_date",
+            "created_at",
+            "updated_at",
+            "quantity",
+            "status",
+            "comment",
+        ]
+        read_only_fields = ["created_at", "updated_at"]
+        created_at = TimestampField(read_only=True)
+        updated_at = TimestampField(read_only=True)
+        expiration_date = TimestampField(read_only=True)
+
+    def create(self, validated_data):
+        user = self.context["request"].user
+        validated_data["account"] = user.iaso_profile.account
+
+        return super().create(validated_data)
+
+
+class HassVaccineAuthorizationsPermissions(permissions.BasePermission):
```

Comment:
I'm wondering if we shouldnt generalize this to a Read / Write generic Auth Permission class that takes to basis permissions (read and write) or if it's not already there somewhere ?